__tests__/focus-traversal.test.tsx

```diff
@@ -34,3 +34,42 @@ it('updates traversals', async () => {
   await findByText('count: 6,7')
   await findByText('bigAtom: [{"a":6},{},{"a":7}]')
 })
+
+type BillingData = {
+  id: string
+}
+
+type CustomerData = {
+  id: string
+  billing: BillingData[]
+  someOtherData: string
+}
+
+it('typescript should accept "undefined" as valid value for traversals', async () => {
+  const customerListListAtom = atom<CustomerData[][]>([])
+
+  const nonEmptyCustomerListAtom = focusAtom(customerListListAtom, (optic) =>
+    optic.find((el) => el.length > 0)
+  )
+
+  focusAtom(nonEmptyCustomerListAtom, (optic) => {
+    const result = optic.elems()
+    return result
+  })
+})
+
+it('should work with promise based atoms with "undefined" value', async () => {
+  const customerBaseAtom = atom<CustomerData[] | undefined>(undefined)
+
+  const asyncCustomerDataAtom = atom(
+    async (get) => get(customerBaseAtom),
+    (_, set, nextValue: CustomerData[]) => {
+      set(customerBaseAtom, nextValue)
+    }
+  )
+
+  focusAtom(asyncCustomerDataAtom, (optic) => {
+    const result = optic.elems()
+    return result
+  })
+})
```

Comment:
@dai-shi I think these tests are missing the `expect` assertion, is that intentional?

Reply:
It comes with an internal type, but let me try again.

Reply:
44d5ae2 added something, but module resolver complains. Any workaround?

Reply:
Seems like Traversals are not designed to support `undefined`. It has zero` or `more` _focuses_ 🤔 so the return type must always be an array.

I found this in [their docs](https://akheron.github.io/optics-ts/reference-mc/#collect).
<img width="791" alt="image" src="https://user-images.githubusercontent.com/14841132/214732455-5102756b-013b-45aa-b158-731aede243e1.png">

So to fix it users would technically need to provide an empty array for undefined values?

Something like this?

```diff
it('should work with promise based atoms with "undefined" value', async () => {
  ...
  const focusedPromiseAtom = focusAtom(asyncCustomerDataAtom, (optic) => {
-     const result = optic.elems()
+    const result = optic.valueOr([] as CustomerData[]).elems()
      return result
  })

  expectTypeOf(focusedPromiseAtom).toMatchTypeOf<
-      NotAnArrayType<CustomerData[] | undefined>[],
-      SetStateAction<NotAnArrayType<CustomerData[] | undefined>>,
-      void
+    WritableAtom<CustomerData[], SetStateAction<CustomerData>, void>
  >()
})
```

What do you think?

Reply:
Sounds reasonable!
Now, I'm pretty confident that you understand optics-ts better than I do. 😄 

Fixed in 729e174.

Reply:
haha, nice!! glad we made it work!

Could you help me release this too?

Reply:
Sure, I will release it.